src/Itinero/Router.cs

```diff
@@ -1320,13 +1320,24 @@ public sealed override Result<bool> TryCheckConnectivity(IProfileInstance profil
                         }
                         weights = algorithm.Weights;
                     }
-                    //else if (contracted.HasNodeBasedGraph &&
-                    //   contracted.NodeBasedIsEdgedBased)
-                    //{ // use vertex-based graph for edge-based routing.
-                    //    weights = Itinero.Algorithms.Contracted.Dual.RouterExtensions.CalculateManyToMany(contracted,
-                    //        _db, profileInstance.Profile,
-                    //        weightHandler, sources, targets, maxSearch, cancellationToken).Value;
-                    //}
+                    else if (contracted.HasNodeBasedGraph &&
```

Comment:
Added back to attempt to be more correct however it now uses the `Itinero.Algorithms.Default.ManyToMany` as seen below